### DISCUSSION TITLE
porta 80

### DISCUSSION THREAD

**Discussion by Lino99** [indomus/forum#1041] [Networking]
Salve a  tutti
ho seguito la guida:
Collegarsi da remoto a Home Assistant Core installato su Raspberry Pi OS

ma il mio fornitore internet, fastweb, non mi permette di fare il forwarding della porta 80.

Soluzioni, oltre a cambiare fornitore?

**Comment by snakuzzo** [comment]
A che ti serve la porta 80 ?
Devi utilizzare la http-01 challenge per rilascio/rinnovo certificati ? Se vuoi puoi usare la TLS-alpn challenge sulla 443 (a meno che il tuo provider non blocchi pure quella).
Se usi dehydrated con DNS Challenge non hai bisogno né della porta 80 né della 443
Se usi un reverse proxy come Caddy...stesso discorso...non userai la challenge http, ma la dns

**Reply by Lino99** [reply]
Grazie risolto usando DNS Challenge, purtroppo dovrò aggiornare le chiavi manualmente.

**Reply by snakuzzo** [reply]
Quali chiavi?
Non ho capito cosa devi fare manualmente

**Reply by Lino99** [reply]
Scusa rinnovo certificato, non chiavi

**Reply by snakuzzo** [reply]
non ti sto capendo...con la DNS Challenge si aggiornano in automatico, perché dici che devi fare tutto manualmente ? :D

**Reply by Lino99** [reply]
Premesso che sto utilizzando certbot e il comando che ho dato è stato:
_sudo certbot certonly --manual --preferred-challenges dns_

Mi ha mostrato una chiave che ho poi inserito come record sul DNS

e infine

_Your cert will expire on 2022-02-15. To obtain a new or tweaked
   version of this certificate in the future, simply run certbot
   again. To non-interactively renew *all* of your certificates, run
   "certbot renew"_

Dove, presumo, mi darà una chiave diversa da inserire nel DNS

**Reply by snakuzzo** [reply]
Ti consiglio di utilizzare un reverse proxy tipo Caddy che ti gestisca tutto in automatico
